scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -118,261 +79,409 @@ class EscapeHatch(
+          filters += EscapeFilter(matcher, rulePos, start, Some(end))
+        }
+        for ((rule, rulePos) <- matchOne) {
+          val unprefixedRuleName = rule.stripPrefix(OptionalRulePrefix)
+          val matcher = FilterMatcher(unprefixedRuleName)
+          filters += EscapeFilter(matcher, rulePos, start, Some(end))
+        }
 
-  private val unusedScalafixSupression = RuleName("UnusedScalafixSupression")
+        builder += (start -> filters.result())
+      }
 
-  private val unusedEnableWarning =
-    LintCategory
-      .warning(
-        "Enable",
-        "This comment would enable a rule that was not disabled (eg: typo in the rules)"
-      )
-      .withOwner(unusedScalafixSupression)
-
-  private val FilterDisable = "\\s?scalafix:off\\s?(.*)".r
-  private val FilterEnable = "\\s?scalafix:on\\s?(.*)".r
-  private val FilterExpression = "\\s?scalafix:ok\\s?(.*)".r
-
-  private def splitRules(rules: String): Set[String] =
-    rules.split("\\s+").toSet
-
-  def apply(tree: Tree, associatedComments: AssociatedComments): EscapeHatch = {
-    val escapes = List.newBuilder[Escape]
-    val unusedEnable = List.newBuilder[LintMessage]
-    val visitedFilterExpression = mutable.Set.empty[AnchorPosition]
-
-    var currentlyDisabledRules = Set.empty[String]
-    def trackUnusedRules(
-        rules: String,
-        toogle: Toogle,
-        anchorPosition: AnchorPosition): Unit = {
-      toogle match {
-        case Toogle.Disable =>
-          currentlyDisabledRules = currentlyDisabledRules ++ splitRules(rules)
-
-        case Toogle.Enable =>
-          val enabledRules = splitRules(rules)
-          val enabledNotDisabled = enabledRules -- currentlyDisabledRules
-
-          enabledNotDisabled.foreach(
-            rule => unusedEnable += unusedEnableWarning.at(anchorPosition.value)
-          )
+      tree.foreach {
+        case t @ Defn.Class(mods, _, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Object(mods, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Trait(mods, _, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Type(mods, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Def(mods, _, _, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Val(mods, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Defn.Var(mods, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Term.Param(mods, _, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Ctor.Primary(mods, _, _) if hasSuppressWarnings(mods) =>
+          addAnnotatedEscape(t, mods)
+
+        case t @ Ctor.Secondary(mods, _, _, _, _)
```

Comment:
Missing `Pkg.Object`

Reply:
done. However, the annotation is not permitted on a package object